service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportRequestUtils.java

```diff
@@ -23,35 +32,48 @@ public static DistributionEnvelopeDocument extractDistributionEnvelope(Element d
         }
     }
 
-    public static POCDMT000002UK01ClinicalDocument1 extractClinicalDocument(DistributionEnvelopeDocument envelopedDocument)
+    public static List<POCDMT000002UK01ClinicalDocument1> extractClinicalDocument(DistributionEnvelopeDocument envelopedDocument)
         throws ItkXmlException {
-        POCDMT000002UK01ClinicalDocument1 clinicalDocument;
+        List<POCDMT000002UK01ClinicalDocument1> clinicalDocuments = new ArrayList<>();
         try {
-            clinicalDocument = ClinicalDocumentDocument1.Factory
-                .parse(findClinicalDoc(envelopedDocument))
-                .getClinicalDocument();
+            List<Node> nodes = findClinicalDoc(envelopedDocument);
+            for (Node node : nodes) {
+                clinicalDocuments.add(ClinicalDocumentDocument1.Factory
+                    .parse(node)
+                    .getClinicalDocument());
+            }
         } catch (XmlException e) {
             throw new ItkXmlException("Clinical document missing from payload", e.getMessage(), e);
         }
 
-        return clinicalDocument;
+        return clinicalDocuments;
     }
 
-    private static Node findClinicalDoc(DistributionEnvelopeDocument envelopedDocument)
+    private static List<Node> findClinicalDoc(DistributionEnvelopeDocument envelopedDocument)
         throws XmlException {
-        NodeList childNodes = envelopedDocument.getDistributionEnvelope()
+        PayloadType[] payloads = Arrays.stream(envelopedDocument.getDistributionEnvelope()
             .getPayloads()
-            .getPayloadArray(0)
-            .getDomNode()
-            .getChildNodes();
+            .getPayloadArray()).toArray(PayloadType[]::new);
 
-        for (int i = 0; i < childNodes.getLength(); i++) {
-            Node node = childNodes.item(i);
+        List<NodeList> nodeListsList =
```

Comment:
I think this stream can be joined to the stream above and return List<NodeList>.